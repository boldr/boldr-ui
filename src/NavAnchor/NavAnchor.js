/* @flow */
import React, { Component } from 'react';
import classnames from 'classnames';
import NavLink from 'react-router-dom/NavLink';

type Props = {
  href: string,
  className: string,
  label: string,
  children: any,
  onClick: () => void,
  icon: any,
};

const NavAnchor = (props: Props) => {
  const { href, className, label, onClick, icon } = props;
  return (
    <NavLink to={href} className="boldrui-link" onClick={onClick} title={label} {...props}>
      {props.icon ? <i className={classnames('fa', props.icon)} /> : null}
      {label}
    </NavLink>
  );
};

export default NavAnchor;
