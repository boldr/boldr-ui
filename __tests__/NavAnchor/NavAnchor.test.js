import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import NavAnchor from '../../src/NavAnchor';

describe('<NavAnchor />', () => {
  const wrapper = shallow(<NavAnchor href="/" title="Home" onClick={() => {}} />);
  it('renders snapshot', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
