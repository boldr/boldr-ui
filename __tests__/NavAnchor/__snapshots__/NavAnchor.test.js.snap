// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavAnchor /> renders snapshot 1`] = `
<NavLink
  activeClassName="active"
  className="boldrui-link"
  href="/"
  onClick={[Function]}
  title="Home"
  to="/"
/>
`;
